internal/servers/controller/handlers/authmethods/authmethod_service.go

```diff
@@ -446,51 +533,53 @@ func (s Service) createInRepo(ctx context.Context, scopeId string, item *pb.Auth
 		}
 		out = am
 	}
-	return toAuthMethodProto(out)
+	return out, nil
 }
 
-func (s Service) updateInRepo(ctx context.Context, scopeId string, req *pbs.UpdateAuthMethodRequest) (*pb.AuthMethod, error) {
+func (s Service) updateInRepo(ctx context.Context, scopeId string, req *pbs.UpdateAuthMethodRequest) (auth.AuthMethod, bool, error) {
 	const op = "authmethods.(Service).updateInRepo"
-	storageToWire := toAuthMethodProto
-	var out auth.AuthMethod
+
+	var am auth.AuthMethod
+	var dryRun bool
+
 	switch auth.SubtypeFromId(req.GetId()) {
 	case auth.PasswordSubtype:
-		am, err := s.updatePwInRepo(ctx, scopeId, req.GetId(), req.GetUpdateMask().GetPaths(), req.GetItem())
+		pam, err := s.updatePwInRepo(ctx, scopeId, req.GetId(), req.GetUpdateMask().GetPaths(), req.GetItem())
 		if err != nil {
-			return nil, errors.Wrap(err, op)
+			return nil, false, errors.Wrap(err, op)
 		}
-		if am == nil {
-			return nil, handlers.ApiErrorWithCodeAndMessage(codes.Internal, "Unable to update auth method but no error returned from repository.")
+		if pam == nil {
+			return nil, false, handlers.ApiErrorWithCodeAndMessage(codes.Internal, "Unable to update auth method but no error returned from repository.")
 		}
-		out = am
+		am = pam
+
 	case auth.OidcSubtype:
-		am, dryRun, err := s.updateOidcInRepo(ctx, scopeId, req)
+		oam, dr, err := s.updateOidcInRepo(ctx, scopeId, req)
 		if err != nil {
-			return nil, errors.Wrap(err, op)
+			return nil, false, errors.Wrap(err, op)
 		}
-		if am == nil {
-			return nil, handlers.ApiErrorWithCodeAndMessage(codes.Internal, "Unable to update auth method but no error returned from repository.")
+		if oam == nil {
+			return nil, false, handlers.ApiErrorWithCodeAndMessage(codes.Internal, "Unable to update auth method but no error returned from repository.")
 		}
-		if dryRun {
-			storageToWire = func(in auth.AuthMethod, opt ...handlers.Option) (*pb.AuthMethod, error) {
-				am, err := toAuthMethodProto(in)
-				if err != nil {
-					return nil, errors.Wrap(err, op)
-				}
-				attrs := &pb.OidcAuthMethodAttributes{}
-				if err := handlers.StructToProto(am.GetAttributes(), attrs); err != nil {
-					return nil, errors.Wrap(err, op, errors.WithMsg("can't convert from attribute struct to proto"))
-				}
-				attrs.DryRun = true
-				if am.Attributes, err = handlers.ProtoToStruct(attrs); err != nil {
-					return nil, errors.Wrap(err, op, errors.WithMsg("can't convert from attribute proto to struct"))
-				}
-				return am, nil
+		if dr {
```

Comment:
This whole code block doesn't seem relevant anymore since we do not use any of the built wire proto.

Reply:
I am honestly not sure what the dry run stuff was so I'm happy to take it out if you and Jim concur but for the moment it works in its updated form.

Reply:
Removed.